### PR TITLE
CI: devel supports Fedora 39, and no longer Fedora 38

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Ref: https://github.com/ansible-collections/news-for-maintainers/issues/63

##### ISSUE TYPE
- Test Pull Request

##### COMPONENT NAME
CI

### CODE REVIEW

File: .azure-pipelines/azure-pipelines.yml [hunk 1]
```diff
@@ -287,6 +287,8 @@ stages:
         parameters:
           testFormat: 2.16/linux/{0}
           targets:
+            - name: Fedora 38
```
Comment (Akasurde) [inline .azure-pipelines/azure-pipelines.yml:290]: Shouldn't this be `Fedora 39`?

Comment (felixfontein) [reply .azure-pipelines/azure-pipelines.yml:290]: No, since this is `fedora38` for stable-2.16.

**Comment by felixfontein** [comment]
The failing `mail` tests are potentially related to #4656.

**Comment by felixfontein** [comment]
The FreeBSD failure is unrelated, see #7543.

**Pull request merged**